src/Core/IFolderWorkspaceService.cs

```diff
@@ -39,6 +39,6 @@ public interface IFolderWorkspaceService
         /// <summary>
         /// Returns the list of files in a folder view
         /// </summary>
-        IEnumerable<string> ListFiles();
+        ICollection<string> ListFiles();
```

Comment:
Does the consumer need to be able to Add and remove items from the list? If not, this should be an array instead

Reply:
It would constrain the implementation too much

Reply:
Changed to IReadOnlyCollection

Reply:
IReadOnlyCollection is fine. 
We're free to change any API anytime we want. So having `string[]` doesn't introduce any problems on the long run.